wa/workloads/uibench/__init__.py

```diff
@@ -0,0 +1,44 @@
+#    Copyright 2013-2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+
+from wa import Parameter, ApkWorkload
+
+
+class Uibench(ApkWorkload):
+
+    name = 'uibench'
+    description = 'Runs a particular activity of the UIBench workload suite.'
+    package_names = ['com.android.test.uibench']
+    loading_time = 1
+
+    parameters = [
+        Parameter('activity', kind=str,
+                  description='The UIBench activity to be run. Please not that '
+                  'the default value of this parameter will result in UIBench '
+                  'opening in its workload selection menu.'),
+        Parameter('duration', kind=int, default=10, description='As activities '
+                  'do not finish, this workload will terminate UIBench after '
+                  'the given duration.'),
+    ]
+
+    def setup(self, context):
+        if self.activity is not None:
+            if self.activity not in self.apk.apk_info.activities:
+                msg = 'Unknown activity "{}". Known activities:\n\t{}'
+                activities = '\n\t'.join(self.apk.apk_info.activities)
+                raise ValueError(msg.format(self.activity, activities))
+            self.apk.apk_info.activity = '.{}'.format(self.activity)
+        super(Uibench, self).setup(context)
+        time.sleep(self.duration)
```

Comment:
uibench might not be the most familiar workload to developers.  Could we be more helpful by adding a short description of the workload and provide a url (https://android.googlesource.com/platform/frameworks/base.git/+/master/tests/UiBench/) for reference?

Reply:
What about the current version?